internal/config/config.go

```diff
@@ -0,0 +1,103 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/spf13/viper"
+)
+
+type JWTInfos struct {
+	Username       string
+	ExpirationTime time.Time
+}
+
+// Set and return auth config as config file
+func authConfigFile() string {
+	// Define config path
+	authConfigFile := filepath.Join(os.Getenv("HOME"), ".config", "paastech", "auth.yaml")
+	viper.SetConfigFile(authConfigFile)
+	viper.ReadInConfig()
+	return authConfigFile
+}
+
+// Load the auth config (jwt) from $HOME/.config/paastech/auth.yaml
+func LoadAuthConfig() error {
+	// Set auth config file as config
+	authConfigFile := authConfigFile()
+
+	// Check if the auth.yaml file exists, if not, create it
+	if _, err := os.Stat(authConfigFile); os.IsNotExist(err) {
+		err = createAuthConfig(authConfigFile)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Read config from file
+	err := viper.ReadInConfig()
+	if err != nil {
+		return errors.New("Failed to read user config file")
+	}
+
+	return nil
+}
+
+// Set the jwt in config file
+func SetAuth(server string, jwt string) {
+	// Set auth config file as config
+	authConfigFile()
+
+	// Change server and jwt value in config file
+	viper.Set("server", server)
+	viper.Set("jwt", jwt)
+	viper.WriteConfig()
+}
+
+// Extract payload from jwt and return it as UserInfos struct
+func ExtractJWTInfos() (JWTInfos, error) {
+	// Set auth config file as config
+	authConfigFile()
+
+	// Get jwt from config
+	tokenString := viper.GetString("jwt")
+	if tokenString == "" {
+		return JWTInfos{}, errors.New("Not logged in")
+	}
+
+	// Parse payload
+	token, _, err := new(jwt.Parser).ParseUnverified(tokenString, jwt.MapClaims{})
```

Comment:
In doc, they say to not use [ParseUnverified](https://pkg.go.dev/github.com/golang-jwt/jwt/v4#Parser.ParseUnverified) because it doesn't check the signature

Shouldn't you use `Parse` instead